core/src/main/java/jenkins/widgets/HistoryPageFilter.java

```diff
@@ -101,85 +105,100 @@ public void setSearchString(@Nonnull String searchString) {
     /**
      * Add build items to the History page.
      *
-     * @param items The items to be added. Assumes the list of items are in descending queue ID order i.e. newest first.
+     * @param runItems The items to be added. Assumes the list of items are in descending queue ID order i.e. newest first.
      */
-    public void add(@Nonnull List<T> items) {
-        if (items.isEmpty()) {
+    public void add(@Nonnull Iterable<T> runItems) {
+        addInternal(runItems);
+    }
+
+    /**
+     * Add run items and queued items to the History page.
+     *
+     * @param runItems The items to be added. Assumes the list of items are in descending queue ID order i.e. newest first.
+     * @param queueItems The queue items to be added.
+     */
+    public void add(@Nonnull Iterable<T> runItems, @Nonnull List<Queue.Item> queueItems) {
```

Comment:
@dennisjlee No explicit docs that I'm aware of. By default, everything that's `public` would be available to plugins. If that's intended and you're writing a stable API, that's fine. Otherwise, annotate `@Restricted` and either `DoNotUse.class` (for stuff only used e.g. from Jelly/Groovy views), and `NoExternalUse.class` (if not for use from plugins).

As the API author, it's really up to you.